### PR TITLE
Support building without libevent

### PR DESCRIPTION
**Pull request by luke-jr**
Without libevent, bitcoind and bitcoin-cli cannot be built, and bitcoin-qt no longer supports -server, -rest, or -torcontrol features (although the debug window still works).

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -742,6 +759,77 @@ else
   fi
 fi
 
+dnl libevent check
```
Comment (jonasschnelli) [inline configure.ac]: would using a m4 marco make sense here. Something like https://github.com/digitalbitbox/dbb-app/blob/master/build-aux/m4/ax_libevent.m4?

**Comment by jonasschnelli** [comment]
Concept ACK.

IMO the only use cases that makes sense:
- Building bitcoin-qt without the -server option and therefore without the libevent dependency.

Not sure if building bitcoind without a RPC server results in something people could use.
Building bitcoin-tx without bitcoind is probably an edge case.

File: configure.ac (hunk 1 above)
Comment (luke-jr) [inline configure.ac]: Probably, but IMO outside the scope of this PR.

**Comment by luke-jr** [comment]
> Not sure if building bitcoind without a RPC server results in something people could use.

Right now, this PR has the following behaviour:

* --without-libevent --enable-daemon: error
* --without-libevent: do not build bitcoind
* --disable-daemon: use libevent if available
* (neither flag): error if libevent is not available

> Building bitcoin-tx without bitcoind is probably an edge case.

It is the *typical* case for at least Gentoo users.

**Comment by laanwj** [comment]
Ping @theuni

**Comment by laanwj** [comment]
I just realized:  don't forget that work is underway to use libevent for the P2P code. So this would only be a short-lived change.

**Comment by luke-jr** [comment]
Hmm, good point. I'd still appreciate review on this, even if it doesn't get merged, since I plan to include it in Bitcoin LJR 0.12.

**Comment by laanwj** [comment]
The code changes look OK to me

**Comment by laanwj** [comment]
Closing this; sure, you could use it as a temporary measure, but it doesn't make sense for us given @theuni's work in progress.
(eventually this will be a way of building just bitcoin-tx, which is a very much an edge case)

**Comment by luke-jr** [comment]
Please reopen this. It occurs to me that libbitcoinconsensus and bitcoin-tx will never need libevent.

**Comment by jtimon** [comment]
Whether libconsensus needs to support concurrency internally or not (like libsecp256k1 doesn't ) is purely a design choice. My preferred choice is that concurrency is implemented by the caller (I highly doubt libbitcoin would use it otherwise, for example ). See #7566 compared to #7575 for what I mean by decoupling libconsensus from concurrency and storage.

**Comment by luke-jr** [comment]
In addition to libbitcoinconsensus/bitcoin-tx, we've now gone through two major releases where this would have been useful.

@jtimon libevent isn't concurrency; it's networking.

**Comment by jtimon** [comment]
Sorry for the previous comment. It was not relevant to this PR at all. Please ignore it for this PR.

**Pull request closed without merging**